services/QuillLMS/app/models/classroom.rb

```
@@ -178,6 +183,8 @@ def classroom_type_for_segment
       'Google Classroom'
     elsif clever_id
       'Clever'
+    elsif canvas_classroom
```

Using a switch statement here, but not enforcing the existence of one-and-only-one property, could lead to hard-to-diagnose bugs or unexpected behavior downstream. 

Let's throw an exception here if, say, a Classroom has a google ID AND a canvas classroom. 

Or you could enforce this integrity in a different way, I don't have a strong preference about the precise implementation.

---

The problem here is that we already have some records with both.  The precedence for now is to give google classroom priority.